gix-pack/src/cache/delta/traverse/util.rs

```diff
@@ -1,28 +1,49 @@
+use std::marker::PhantomData;
+
 use crate::cache::delta::Item;
 
-pub struct ItemSliceSend<T>(pub *mut [T])
+pub(crate) struct ItemSliceSend<'a, T>
+where
+    T: Send,
+{
+    items: *mut T,
+    phantom: PhantomData<&'a T>,
```

Comment:
Nice! The `phantom` has the sole purpose of maintaining the lifetime of the input slice, even though internally it's reduced to a pointer.

This also means the bounds-check isn't happening anymore upon access, but then again, it's also not needed anymore as it's known that all indices fit within the slice.